### DISCUSSION TITLE
Meaning of certain termination criteria

### DISCUSSION THREAD

**Discussion by ThomasEwaldETH** [anyoptimization/pymoo#400] [Q&A] [answered]
Hello everyone!
First, thanks a lot for this great optimization toolbox, which helps me a lot right now.

However, I have a question regarding the termination criteria. Due to [homepage ](https://pymoo.org/interface/termination.html)information I use the following setup for my problem (copied the code from there):

    Termination(RobustTermination):
        """Termination class"""

        def __init__(self) -> None:
            super().__init__(
                period=20,
                termination=MultiObjectiveSpaceTermination(
                    tol=0.005,
                    n_skip=5,
                ),
            )

Hereby, I have problems with understanding the termination criteria. The description on the homepage says:

- tol: What is the tolerance in the objective space on average. If the value is below this bound, we terminate.
- n_last: To make the criterion more robust, we consider the last n generations and take the maximum. This considers the worst case in a window.
- n_max_gen: As a fallback, the generation number can be used. For some problems, the termination criterion might not be reached; however, an upper bound for generations can be defined to stop in that case.
- nth_gen: Defines whenever the termination criterion is calculated by default, every 10th generation.

Unfortunately, those are not the same as given in the example. So, what are 'period', and 'n_skip' (is 'n_skip' the same as 'n_last')? The rest of the bullet point list seems not to work in the classes that are used in the example.

More generally: I assume, that Objective Space Tolerance is the right one for me to choose, but I don't see where I can find detailled information on how to control this tolerance.

Any help will be greatly appreciated!
Best regards and a very nice day
Thomas

**Comment by blankjul** [comment, marked as answer]
You are right the documentation is not up to date here. Can you create an issue that it gets some priority in the next version to be updated?

`period` means the number of last generations (before called `n_last`) to be considered. `n_skip` means how many generations are skipped until termination is checked again.

You can also directly use the default termination by not providing any parameter (it can be better to play with some parameters though).

**Reply by ThomasEwaldETH** [reply]
Hey, and thanks a lot for the quick answer!

I can work with this information already. I set n_skip to 0 (zero) and try period with five (5), to be sure.

However, I have a small follow-up question on the Termination class. I realized that the different classes are not the same in terms of their arguments I have to provide to the constructor (I had a look at all constructors in question). Especially, my choice seems to do not provide a guaranteed cancellation after a certain number of iterations (n_max_gen). Is this on purpose?
Otherwise, also thanks for the hint to create an issue, I will do that soon.

Overall, thanks again for this great optimization module!
Best regards, Thomas
